test/distributed/fsdp/test_fsdp_fx.py

```diff
@@ -0,0 +1,113 @@
+# Owner(s): ["oncall: distributed"]
+
+import torch
+from torch.distributed.fsdp._symbolic_trace import _init_execution_info, _patch_tracer
+from torch.testing._internal.common_fsdp import FSDPTest
+from torch.testing._internal.common_utils import (
+    instantiate_parametrized_tests,
+    run_tests,
+)
+
+
+class Model(torch.nn.Module):
```

Comment:
what does the tracer do if there's conditional execution / use of parameters?

Reply:
Hi Rohan, I believe whether the tracer could work successfully depends on two cases:

- if the branch is based on some constants in the model, the tracer could still work property, see https://github.com/pytorch/pytorch/blob/master/test/fx/test_fx_param_shape_control_flow.py
- if the branch is based on the value of some inputs to the forward function, then the only way to get the tracer working properly is to add these inputs in `concrete_args`, which is just to tell `torch.fx` that these inputs are fixed, so there is a fixed path in the branch.